addons/product/models/product_template.py

```diff
@@ -538,7 +538,31 @@ def copy(self, default=None):
             default = {}
         if 'name' not in default:
             default['name'] = _("%s (copy)", self.name)
-        return super(ProductTemplate, self).copy(default=default)
+
+        res = super(ProductTemplate, self).copy(default=default)
+
+        # Since we don't copy the attribute lines directly, we need to match the extra price
+        # and create the exclusions of the newly created attribute line values.
+        # This matching expects these values to be uniquely determined by the attribute
+        # value. If two lines uses the same attribute value, it will not work properly.
+        copied_ptavs = res.attribute_line_ids.product_template_value_ids
+        for ptav in self.attribute_line_ids.product_template_value_ids:
+            if ptav.price_extra:
+                copied_ptavs.filtered(
+                    lambda x: x.product_attribute_value_id == ptav.product_attribute_value_id
+                ).price_extra = ptav.price_extra
+            for excl in ptav.exclude_for:
+                new_exclusion = self.env['product.template.attribute.exclusion'].create({
+                    'product_tmpl_id': res.id if excl.product_tmpl_id == self else excl.product_tmpl_id.id,
+                    'value_ids': copied_ptavs.filtered(
+                        lambda x: x.product_attribute_value_id in excl.value_ids.product_attribute_value_id
+                    ),
```

Comment:
I am not sure about this change. The records present in `excl.values_ids`, are `product.template.attribute.values` as such, these record are linked to a `product_tmpl_id` and an `attribute_id`. Therefore, using the `values_ids` of the original `product_template` would not be coherent with the rest of the matching (as an other `product_tmpl_id` and an other `attribute_id` are already matched to the exclusion). I think that even if it might not create a bug right away, it might me be the cause of other problematic flows in the future.

Reply:
What about exclusions that have attributes values not included in the product template attributes values?

Reply:
Good catch :)